src/ssh-compute.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as toolCache from '@actions/tool-cache';
+import * as setupGcloud from '@google-github-actions/setup-cloud-sdk';
+import path from 'path';
+import fs from 'fs';
+
+export const GCLOUD_METRICS_ENV_VAR = 'CLOUDSDK_METRICS_ENVIRONMENT';
+export const GCLOUD_METRICS_LABEL = 'github-actions-ssh-compute';
+
+export function parseFlags(flags: string): RegExpMatchArray {
+  return flags.match(/(".*?"|[^"\s=]+)+(?=\s*|\s*$)/g)!; // Split on space or "=" if not in quotes
+}
+
+/**
+ * Executes the main action. It includes the main business logic and is the
+ * primary entry point. It is documented inline.
+ */
+export async function run(): Promise<void> {
+  core.exportVariable(GCLOUD_METRICS_ENV_VAR, GCLOUD_METRICS_LABEL);
+  try {
+    // Get inputs
+    // Core inputs
+    let instanceName = core.getInput('instance_name');
+    const zone = core.getInput('zone');
+    const user = core.getInput('user');
+    const container = core.getInput('container');
+    const sshKeyFilePath = core.getInput('ssh_key_file_path') || '/home/runner/.ssh/google_compute_engine';
+    const sshKeyExpireAfter = core.getInput('ssh_key_expire_after');
+    const sshArgs = core.getInput('ssh_args');
+    let command = core.getInput('command');
+    const entrypoint = core.getInput('entrypoint');
+    const credentials = core.getInput('credentials');
+    let projectId = core.getInput('project_id');
+    let gcloudVersion = core.getInput('gcloud_version');
+    const generateSshKeys = core.getInput('generate_ssh_keys');
+  
+    // Flags
+    const internalIp = core.getInput('internal_ip');
+    const tunnelThroughIap = core.getInput('tunnel_through_iap');
+    const flags = core.getInput('flags');
+    const installBeta = false; // Flag for installing gcloud beta components
```

Comment:
Any reason why not just use `beta` by default?

Reply:
Change to true.